frontend/pages/hosts/details/DeviceUserPage/DeviceUserPage.tsx

```diff
@@ -229,7 +229,7 @@ const DeviceUserPage = ({
   const statusClassName = classnames("status", `status--${host?.status}`);
 
   const renderDeviceUserPage = () => {
-    const failing_policies_count = titleData?.issues;
+    const failingPoliciesCount = host?.issues?.failing_policies_count || 0;
```

Comment:
An easy way we can improve the typing that comes out of `normalizeEmptyValues` would be to use a type assertion when it is called. With that change, you should be able to use `titleData` again.

```ts
  const titleData: Partial<IHost> = normalizeEmptyValues(
    pick(host, [
      "status",
      "issues",
      "memory",
      "cpu_type",
      "os_version",
      "osquery_version",
      "enroll_secret_name",
      "detail_updated_at",
      "percent_disk_space_available",
      "gigs_disk_space_available",
      "team_name",
      "hostname",
    ])
  );
```

Reply:
I think it would be worth including in this PR for the other places it is used too (e.g., `aboutData`, `osQueryData`)

Reply:
Ah good call. Will update.

Reply:
Hmmm. I realized this opens a few other hidden problems. By asserting the type, we're potentially setting up type mismatches. For example, what if `gigs_disk_space_available` is 0? Then we'd have TypeScript thinking that "---" is actually a number, which would lead to some unexpected runtime problems. So it would be better to stick with the union type and only assert after checking the type of the individual fields.